gammapy/spectrum/energy_group_new.py

```diff
@@ -0,0 +1,37 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Spectrum energy bin grouping.
+"""
+import numpy as np
+import astropy.units as u
+
+class FluxPointBins:
+    def __init__(self, indices):
+        self.indices = indices
+
+    def lower_bounds(self):
+        return self.indices[:-1]
+
+    def upper_bounds(self):
+        return self.indices[1:] - 1
+
+    def bin(self, idx):
+        return np.arange(self.lower_bounds[idx],
+                         self.upper_bounds[idx])
+
+
+
+class FluxPointBinMaker:
```

Comment:
I'm a bit perplexed by the ubiquity of these `maker` classes within gammapy. Why doesn't this functionality simply go into the  FluxPointBins class?

This directly implies the next obvious question of why there is a class called `FluxPoints`but no class called `FluxPoint` which just knows its bin edges?

Reply:
I completely agree about Maker classes. To me it makes python look like Java. I would have just made an initialization method to FluxPointBins, but was told that separate maker classes are the house style in gammapy. Is that not the case?

Reply:
Regarding FluxPoint, that adds some complexity. Really this whole class is just a wrapper around a single array of indices. FluxPoint would then be a wrapper around just one index.

Reply:
> Regarding FluxPoint, that adds some complexity. Really this whole class is just a wrapper around a single array of indices. FluxPoint would then be a wrapper around just one index.

I agree. It was a somewhat suggestive question. It seems to me like its a bit of OO overkill in this case.

> To me it makes python look like Java.
Exactly. That's what I was reminded of  :smile:

Reply:
> but was told that separate maker classes are the house style in gammapy. Is that not the case?

I don't know to be honest. Maybe @cdeil can chime in.

Reply:
Sorry, one pair of indices (obviously)